security/Member.php

```diff
@@ -1497,26 +1502,34 @@ public function canView($member = null) {
 	 * Otherwise they'll need ADMIN or CMS_ACCESS_SecurityAdmin permissions
 	 */
 	public function canEdit($member = null) {
-		if(!$member || !(is_a($member, 'Member')) || is_numeric($member)) $member = Member::currentUser();
-
-		// extended access checks
-		$results = $this->extend('canEdit', $member);
-		if($results && is_array($results)) {
-			if(!min($results)) return false;
-			else return true;
+		//get member
+		if(!($member instanceof Member)) {
+			$member = Member::currentUser();
 		}
-
-		// No member found
-		if(!($member && $member->exists())) return false;
-
-		// If the requesting member is not an admin, but has access to manage members,
-		// they still can't edit other members with ADMIN permission.
-		// This is a bit weak, strictly speaking they shouldn't be allowed to
-		// perform any action that could change the password on a member
-		// with "higher" permissions than himself, but thats hard to determine.
+		//check for extensions, we do this first as they can overrule everything
+		$extended = $this->extendedCan(__FUNCTION__, $member);
+		if($extended !== null) {
+			return $extended;
+		}
+		
+		//need to be logged in and/or
+		//most checks below rely on $member being a Member
+		if(!$member) return false;		
+		
+		// HACK: we should not allow for an non-Admin to edit an Admin
 		if(!Permission::checkMember($member, 'ADMIN') && Permission::checkMember($this, 'ADMIN')) return false;
```

Comment:
But yeah, `Please use braces on if conditions.` (I bet you're sick of hearing this from me) :P